apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -572,26 +523,33 @@ mod helpers {
                                     // selection.
                                     let field_output = field_def.ty.inner_named_type();
                                     if to_schema.try_get_type(field_output.clone()).is_none() {
-                                        let field_output =
-                                            self.original_schema.get_type(field_output.clone())?;
-                                        let field_type =
-                                            field_output.get(self.original_schema.schema())?;
-
                                         // We use a fake JSONSelection here so that we can reuse the visitor
                                         // when generating the output types and their required memebers.
-                                        let visitor = ToSchemaVisitor::new(
+                                        let visitor = SchemaVisitor::new(
                                             self.original_schema,
                                             to_schema,
-                                            field_output,
-                                            field_type,
                                             &self.directive_deny_list,
                                         );
                                         let (_, parsed) =
                                             JSONSelection::parse(&sub_selection).map_err(|e| {
                                                 FederationError::internal(format!("could not parse fake selection for sibling field: {e}"))
                                             })?;
 
-                                        visitor.walk(&parsed)?;
+                                        let output_type = match self
+                                            .original_schema
+                                            .get_type(field_output.clone())?
+                                        {
+                                            TypeDefinitionPosition::Object(object) => object,
+
+                                            other => {
+                                                return Err(FederationError::internal(format!("connector output types currently only support object types: found {}", other.type_name())))
+                                            }
+                                        };
+
+                                        visitor.walk((
+                                            output_type,
+                                            parsed.next_subselection().cloned().unwrap(),
```

Comment:
either unwrap->expect or make this conditional on a `Some` value?

Reply:
Expect for now, better errors in a later PR